### PR TITLE
sync #2298

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-devpack-dotnet/issues/466 https://github.com/neo-project/neo-devpack-dotnet/issues/468

### CODE REVIEW

File: src/Neo.SmartContract.Framework/Services/Neo/Crypto.cs [hunk 1]
```diff
@@ -2,39 +2,20 @@ namespace Neo.SmartContract.Framework.Services.Neo
...
-
-                [Syscall("Neo.Crypto.CheckMultisigWithECDsaSecp256r1")]
-                public extern static bool CheckMultiSig(ByteString message, Cryptography.ECC.ECPoint[] pubkey, ByteString[] signature);
-            }
-
-            public static class Secp256k1
-            {
-                [Syscall("Neo.Crypto.VerifyWithECDsaSecp256k1")]
-                public extern static bool Verify(ByteString message, Cryptography.ECC.ECPoint pubkey, ByteString signature);
+            return CryptoLib.ripemd160(CryptoLib.Sha256(value));
```
Comment (vncoelho) [inline src/Neo.SmartContract.Framework/Services/Neo/Crypto.cs:13]: Very good and simple, now the basic one are in CryptoLib.
This opens more possibilities.

**Comment by shargon** [comment]
merge?

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 2]
```diff
@@ -722,14 +723,24 @@ private int ConvertCall(OpCode src, NeoMethod to, List<MethodToken> methodTokens
                 {
                     ConvertPushNumber(10, null, to);        // Push Base
                     Convert1by1(VM.OpCode.SWAP, src, to);   // Swap arguments
-                    Insert1(VM.OpCode.SYSCALL, "", to, BitConverter.GetBytes(ApplicationEngine.System_Binary_Itoa));
+                    ConvertPushNumber(2, null, to);
+                    Convert1by1(VM.OpCode.PACK, src, to);   // PACK
+                    ConvertPushNumber((int)CallFlags.All, null, to); // add CallFlag
+                    ConvertPushString("itoa", null, to);
+                    ConvertPushDataArray(NativeContract.StdLib.Hash.ToArray(), null, to);
+                    Convert1by1(VM.OpCode.SYSCALL, null, to, BitConverter.GetBytes(ApplicationEngine.System_Contract_Call));
```
Comment (erikzhang) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs:731]: I think we should use `CALLT`.

Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: Fixed

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 3]
```diff
@@ -919,15 +922,20 @@ private int ConvertCall(OpCode src, NeoMethod to, List<MethodToken> methodTokens
             return 0;
         }
 
-        private ushort AddMethodToken(List<MethodToken> methodTokens, UInt160 hash, MethodDefinition method, CallFlags flags)
+        private ushort AddMethodToken(List<MethodToken> methodTokens, UInt160 hash, CallFlags flags, MethodDefinition method)
+        {
+            return AddMethodToken(methodTokens, hash, flags, method.Name, (ushort)method.Parameters.Count, method.ReturnType.FullName != FuncExport.Void.FullName);
```
Comment (erikzhang) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs:927]: `GetMethodName(method)`?

Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: Fixed!

File: src/Neo.SmartContract.Framework/Services/Neo/CryptoLib.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    [Contract("0x726cb6e0cd8628a1350a611384688911ab75f51b")]
+    public static class CryptoLib
+    {
+        /// <summary>
+        /// RFC 4492
+        /// </summary>
+        /// <remarks>
+        /// https://tools.ietf.org/html/rfc4492#section-5.1.1
+        /// </remarks>
+        public enum NamedCurve : byte
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Services/Neo/CryptoLib.cs:14]: One file one class?

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/CryptoLib.cs]: Done

File: src/Neo.SmartContract.Framework/Services/Neo/StdLib.cs [hunk 5]
```diff
@@ -1,49 +1,48 @@
...
-    public class Binary
+    [Contract("0xacce6fd80d44e1796aa0c2c625e9e4e0ce39efc0")]
+    public static class StdLib
     {
-        [Syscall("System.Binary.Base64Decode")]
+        public static extern UInt160 Hash { [ContractHash] get; }
+
+        public extern static byte[] Serialize(object source);
+
+        public extern static object Deserialize(byte[] source);
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Services/Neo/StdLib.cs:14]: `ByteString`

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/StdLib.cs]: Done

**Review by erikzhang — APPROVED** [review]

**Pull request merged**